File: 875. Koko Eating Bananas/875. Koko Eating Bananas.java
import java.util.Arrays;

class Solution {

  public int minEatingSpeed(int[] piles, int h) {
    int low = 1;
    int high = Arrays.stream(piles).max().getAsInt();

    while (low < high) {
      int mid = (low + high) / 2;

      int hours = 0;
      for (int pile : piles) {
        hours += pile / mid;
        if (pile % mid != 0) {
          hours++;
        }
      }

      if (hours <= h) {
        high = mid;
      } else {
        low = mid + 1;
      }
    }

    return low;
  }
}

File: 875. Koko Eating Bananas/875. Koko Eating Bananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */
var minEatingSpeed = function (piles, h) {
  let low = 1;
  let high = Math.max(...piles);

  while (low < high) {
    mid = Math.floor((low + high) / 2);

    let hours = 0;
    for (const pile of piles) {
      hours += Math.floor(pile / mid);
      if (pile % mid !== 0) {
        hours++;
      }
    }

    if (hours <= h) {
      high = mid;
    } else {
      low = mid + 1;
    }
  }

  return low;
};
